legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure

```diff
@@ -119,6 +119,25 @@ function meta::relational::mapping::generateInstantiationExecutionNode(sq:meta::
    ]);
 }
 
+function meta::relational::mapping::assertCompatibleTypesInRelationalPropertyMapping(r:RelationalPropertyMapping[0..1]):Boolean[1]
+{
+  let classPropertyPureType = $r.property.genericType.rawType;
+  let ropType = meta::relational::mapping::getRelationalTypeFromRelationalPropertyMapping($r, '');
+  let ropCompatiblePureType = if($ropType->isEmpty(),
+      | [] 
+     ,| dataTypeToCompatiblePureType($ropType->toOne());
+     );
+
+  let ropCompatiblePureTypeGeneralizations = if($ropCompatiblePureType->isEmpty(),
+      | [] 
+     ,| $ropCompatiblePureType->toOne()->meta::pure::functions::meta::generalizations();
+     );
+
+  assert($classPropertyPureType->isEmpty() || $ropCompatiblePureTypeGeneralizations->isEmpty() || 
+  $ropCompatiblePureTypeGeneralizations->filter(f| $f != Any)->exists(m|$classPropertyPureType->toOne()->subTypeOf($m)),
```

Comment:
Make use of _subTypeOf function. IIRC, normal subTypeOf wouldn't work on user models (works only on core models - defined in core repositories).

Also, can the check be simplified? (Isn't $propertyType->_subTypeOf($columnType) enough)

Reply:
Also, can the check be simplified? (Isn't $propertyType->_subTypeOf($columnType) enough) 

-> We are using generalizations of columnType because some compatible types like Date and StrictDate were failing with direct comparison.

Reply:
Also, can the check be simplified? (Isn't $propertyType->_subTypeOf($columnType) enough)

->It can't be simplified as none of the type is supertype of other. So we need generalizations for both.